ui/helpers/utils/install-critical-error-listeners.ts

```diff
@@ -0,0 +1,66 @@
+import type browser from 'webextension-polyfill';
+import { isObject } from 'lodash';
+import { hasProperty } from '@metamask/utils';
+import log from 'loglevel';
+import { METHOD_DISPLAY_STATE_CORRUPTION_ERROR } from '../../../shared/constants/state-corruption';
+import type { ErrorLike } from '../../../shared/constants/errors';
+import { displayStateCorruptionError } from './state-corruption-html';
+
+/**
+ * Connects error listeners to the provided port to handle state corruption errors.
+ * This function listens for messages from the background script and displays
+ * a state corruption error if the appropriate message is received.
+ *
+ * Critical error messages are transferred over a raw browser `Port`, not with
+ * `PortStream` wrapper. We want to be as close to the "metal" as possible here,
+ * it minimize abstractions that could cause further issues.
+ *
+ * @param container - The container element to display the error in.
+ * @param port - The port to listen for messages on.
+ */
+export function installCriticalStartupErrorListeners(
```

Comment:
This change here generalizes the previous "vault corruption" error listener into something generalized for any error. We are undergoing an effort to capture additional start-up related errors and display them to the user right now.